abd-clam/src/core/graph/_graph.rs

```diff
@@ -498,14 +513,18 @@ impl<'a, U: Number> Graph<'a, U> {
     ///
     /// # Returns
     ///
-    /// A reference to the distance matrix, which is a 2D vector of the specified number type.
-    #[must_use]
-    pub const fn distance_matrix(&self) -> &[Vec<U>] {
-        todo!()
-
-        // self.distance_matrix
-        //     .as_ref()
-        //     .expect("Please call `with_distance_matrix` on the Graph before using `distance_matrix`.")
+    /// A `Result` where:
+    /// - `Ok` contains a reference to the distance matrix, which is a 2D vector of the specified number type.
+    /// - `Err` contains an error message if `with_distance_matrix` was not called before using this method.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error message if the distance matrix is not available, indicating that `with_distance_matrix`
+    /// should be called before using this method.
+    pub fn distance_matrix(&self) -> Result<&[Vec<U>], String> {
```

Comment:
These should just return `Option`s instead of `Result`s

Reply:
fixed